main.ts

```diff
@@ -699,21 +699,23 @@ class Translator {
       case ts.SyntaxKind.Parameter:
         var paramDecl = <ts.ParameterDeclaration>node;
         if (paramDecl.dotDotDotToken) this.reportError(node, 'rest parameters are unsupported');
-        if (paramDecl.initializer) this.emit('[');
-        this.visitEachIfPresent(paramDecl.decorators);
-        if (paramDecl.type) {
-          if (paramDecl.name.kind !== ts.SyntaxKind.ObjectBindingPattern) {
-            this.visit(paramDecl.type);
-          } else {
+        if (paramDecl.name.kind === ts.SyntaxKind.ObjectBindingPattern) {
+          this.visitEachIfPresent(paramDecl.decorators);
+          if (paramDecl.type) {
             // TODO(martinprobst): These are currently silently ignored.
             // this.reportError(paramDecl.type, 'types on named parameters are unsupported');
           }
-        }
-        this.visit(paramDecl.name);
-        if (paramDecl.initializer) {
-          this.emit('=');
-          this.visit(paramDecl.initializer);
-          this.emit(']');
+          this.visit(paramDecl.name);
```

Comment:
I think this code should check that the initialiser is just an empty object assignment, and if not, report an error.